unit-tests/src/test/java/org/eclipse/collections/impl/set/mutable/primitive/ImmutableBooleanHashSetTest.java

```diff
@@ -625,4 +625,38 @@ public void intersect()
         ImmutableBooleanSet actual5 = set15.intersect(set25);
         Assert.assertEquals(this.emptySet, actual5);
     }
+
+    @Test
+    public void difference()
+    {
+        ImmutableBooleanSet set11 = this.newWith(true);
+        ImmutableBooleanSet set21 = this.newWith(false);
+        ImmutableBooleanSet actual = set11.difference(set21);
+        Assert.assertEquals(this.trueSet, actual);
+
+        ImmutableBooleanSet set12 = this.newWith(false);
+        ImmutableBooleanSet set22 = this.newWith(false);
+        ImmutableBooleanSet actual2 = set12.difference(set22);
+        Assert.assertEquals(this.emptySet, actual2);
+
+        ImmutableBooleanSet set13 = this.newWith(true);
+        ImmutableBooleanSet set23 = this.newWith(true);
+        ImmutableBooleanSet actual3 = set13.difference(set23);
+        Assert.assertEquals(this.emptySet, actual3);
+
+        ImmutableBooleanSet set14 = this.trueFalseSet;
+        ImmutableBooleanSet set24 = this.newWith();
+        ImmutableBooleanSet actual4 = set14.difference(set24);
+        Assert.assertEquals(this.trueFalseSet, actual4);
+
+        ImmutableBooleanSet set15 = this.newWith();
+        ImmutableBooleanSet set25 = this.trueSet;
+        ImmutableBooleanSet actual5 = set15.difference(set25);
+        Assert.assertEquals(this.emptySet, actual5);
+
```

Comment:
I think we just need to add a case where you have `this.trueFalseSet` and call `difference` with `this.trueFalseSet.`

Reply:
Updated, see lines 642-643.